backend/application/import_observations/parsers/trivy_prometheus/parser.py

```diff
@@ -0,0 +1,170 @@
+        )
+
+        trivy_basic_auth_param = None
+        if trivy_prometheus_basic_auth:
+            trivy_basic_auth_param = (
+                trivy_prometheus_basic_auth_username,
+                trivy_prometheus_basic_auth_password,
+            )
+
+        try:
+            response = requests.get(
+                trivy_prometheus_url,
+                timeout=60,
+                verify=trivy_prometheus_verify_ssl,
+                auth=trivy_basic_auth_param,
+            )
+            response.raise_for_status()
+        except Exception as e:
+            return False, [f"Cannot access Prometheus: {str(e)}"], {}
+
+        return True, [], response.json()
+
+    def check_format(self, import_data) -> tuple[bool, list[str], dict]:
+        try:
+            data = json.load(import_data)
+        except Exception:
+            return False, ["Data is not valid JSON"], {}
+
+        if not data.get("status") == "success":
+            return False, ["Data is not a Prometheus API-Endpoint"], {}
+
+        if not isinstance(data.get("data"), dict) or not isinstance(
+            data.get("data").get("result"), list
+        ):
+            return False, ["Data not in valid Prometheus-Metric Format"], {}
+
+        return True, [], data
+
+    def get_observations(self, data) -> list[Observation]:
+        observations = []
+
+
+        for finding in data.get("data").get("result"):
+            if not finding.get("metric", {}).get("vuln_id", ""):
+                continue
+            origin_component_name = finding.get("metric", {}).get("resource", "")
+            vuln_title = finding.get("metric", {}).get("vuln_title", "")
+            vulnerability_id = finding.get("metric", {}).get("vuln_id", "")
+            cvss3_score = finding.get("metric", {}).get("vuln_score")
+            severity = finding.get("metric", {}).get(
+                "severity", Severity.SEVERITY_UNKOWN
+            )
+            origin_docker_image_registry = finding.get("metric", {}).get("image_registry", "")
+            origin_docker_image_repository = finding.get("metric", {}).get("image_repository", "")
+            origin_docker_image_name = (
+                origin_docker_image_registry
+                + "/"
+                + origin_docker_image_repository
+            )
+            origin_docker_image_tag = finding.get("metric", {}).get("image_tag", "")
+            fixed_version = finding.get("metric", {}).get("fixed_version", "")
+            origin_component_version = finding.get("metric", {}).get("installed_version", "")
+            namespace = finding.get("metric", {}).get("namespace", "")
+            resource_kind = finding.get("metric", {}).get("resource_kind", "")
+            resource_name = finding.get("metric", {}).get("resource_name", "")
+            container_name = finding.get("metric", {}).get("container_name", "")
+            origin_endpoint_url = self.api_configuration.base_url
```

Comment:
The origin_endpoint_url is the URL, where the service with the vulnerability is available. That is most likely not the base_url of the API Configuration.